src/main/java/org/openpnp/model/BoardLocation.java

```diff
@@ -98,6 +106,39 @@ public void setSide(Side side) {
         firePropertyChange("side", oldValue, side);
     }
 
+    public void setBoardStatus(BoardStatus boardstatus) {
+        Object oldValue = this.boardStatus;
+        this.boardStatus = boardstatus;
+        firePropertyChange("boardStatus", oldValue, boardStatus);
+    }
+
+    public void initBoardStatus() {
+    	for (Placement placement : getBoard().getPlacements()) {
```

Comment:
Suggest adding a break for each case. No need to iterate through all the placements once you've found an error.

Reply:
done